hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/SchedulerUtils.java

```diff
@@ -367,6 +367,10 @@ public static boolean checkNodeLabelExpression(Set<String> nodeLabels,
     }
 
     if (labelExpression != null) {
+      if(labelExpression.equals(RMNodeLabelsManager.ANY)) {
```

Comment:
My comment on the equals was to do it inside the for-split-&& below, rather than separately, so as to make "&& *" work like it should, i.e. do nothing. It won't matter for us, but I foresee this being a comment from the community who thinks about future support.